### PR TITLE
#9430: extracting first util lib

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

- Part of #9430
- As part of the monorepo trailblazing, this PR sets up the first extracted utils library, extracting some debugging util that isn't used anywhere else (namely `pixiebrix-app`) and doesn't have any other dependencies
  - This will make it easier for us to properly set up `nx` and catch any potential issues.
- Initial set up with some duplicated jest and tsconfig
  - Follow-up will extract common base configuration to the root, and add linting

## Discussion

- For ease of starting this out, this new utils library is build-less, but we will likely want to move towards each library project producing its compiled js so we can take advantage of `nx`'s build and caching mechanisms. I will look into this in the next set of changes alongside creating a root common ts config.
  - For reference see Nx's example npm workspace project: https://github.com/nrwl/nx-examples/tree/master/libs/shared/product/ui

## Checklist

- [X] Added jest or playwright tests and/or storybook stories

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: applications/browser-extension/src/pageEditor/store/editor/editorSlice.test.ts [hunk 1]
```diff
@@ -178,6 +185,42 @@ describe("Add/Remove Bricks", () => {
     ).toBeArrayOfSize(initialBricks.length + 1);
   });
 
+  test("Add Brick - bad pipeline path error", async () => {
```
Comment (fungairino) [inline applications/browser-extension/src/pageEditor/store/editor/editorSlice.test.ts:188]: Added this test just to ensure there was no regression with this existing debug util code.

File: libraries/utils/src/debugUtils.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ type InvalidPathInformation = {
  * @param path period separated path
  */
 export function getInvalidPath(
-  value: UnknownObject,
+  value: Record<string, unknown>,
```
Comment (fungairino) [inline libraries/utils/src/debugUtils.ts:31]: UnknownObject comes from a global type def, which we might eventually want to move to the root as well.

Comment (twschiller) [reply libraries/utils/src/debugUtils.ts:31]: We might also consider using typefest's UnknownRecord instead: https://github.com/sindresorhus/type-fest/blob/main/source/unknown-record.d.ts. There's is probably technically more accurate — ours only supports string keys vs. strings/symbols

File: libraries/utils/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  // You can see the full details at https://github.com/sindresorhus/tsconfig/blob/main/tsconfig.json
+  // Note: `strict: true` enables many flags that aren’t explicitly listed in that file
+  "extends": "@sindresorhus/tsconfig",
+  "compilerOptions": {
+    "sourceMap": true,
+    "module": "esnext",
+    "target": "es2023",
```
Comment (twschiller) [inline libraries/utils/tsconfig.json:8]: Should we consider defining a config these can extend from? E.g., in some libraries we might be able to enforce stricter rules (e.g., `noUnusedParameters`)

**Review by twschiller — APPROVED** [review]

File: libraries/utils/src/debugUtils.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ type InvalidPathInformation = {
  * @param path period separated path
  */
 export function getInvalidPath(
```
Comment (twschiller) [inline libraries/utils/src/debugUtils.ts:30]: As an aside, given this method uses Formik's getIn vs. lodash's we might consider moving it to a formikDebugUtils module or similar in the future

File: libraries/utils/tsconfig.json (hunk 3 above)
Comment (fungairino) [reply libraries/utils/tsconfig.json:8]: Yes, I think we should define a base tsconfig. That's the follow-up I mentioned in the PR description

**Pull request merged**